tests/unit/file-handler.js

```diff
@@ -83,13 +84,39 @@ suite('file handler Suite', () => {
     const sharedFileUri = { path: '.vscode/settings.shared.json' };
     const localFileUri = { path: '.vscode/settings.local.json' };
     const mergeConfigFiles = fileHandler.mergeConfigFiles;
-    const sharedConfig = { foo: false };
-    const localConfig = { foo: true, 'window.zoomLevel': 1, baz: false };
-    const vscodeConfig = { foo: 'abc', 'window.zoomLevel': 0, bar: 'def' };
+    const sharedConfig = {
+      foo: false,
+      'editor.rulers': [100],
+      '[typescript]': {
+        'editor.dragAndDrop': false,
+        'editor.tabCompletion': 'on',
+      },
+    };
+    const localConfig = {
+      foo: true,
+      'window.zoomLevel': 1,
+      'editor.rulers': [80],
+      '[typescript]': {
+        'editor.dragAndDrop': true,
+        'editor.autoIndent': false,
+      },
+      baz: false,
+    };
+    const vscodeConfig = {
+      foo: 'abc',
+      'window.zoomLevel': 0,
+      bar: 'def',
+    };
     const expConfig = {
       foo: true,
       'window.zoomLevel': 1,
       bar: 'def',
+      'editor.rulers': [100, 80],
```

Comment:
I wonder if the array merge of values is going to be the desired behavior in this particular context :thinking:

Don't get me wrong, I think it's correct in the technical, literal sense of deep merging, but it seems like it would break the ability to fully override/set any local keys if there's a corresponding key in the respective shared file.

I'm not all that familiar with the `deepmerge` lib but I wonder if its API surface accepts any args that allow this to be controlled (which we could surface up to users of the extension via a new extension config option)?

Reply:
Should've bothered reading their docs before posting - https://github.com/TehShrike/deepmerge#arraymerge

I'd be good merging this as-is, but I'd want to expose the array merge behavior to end users via a new config option before releasing. If you're up for doing that as part of this PR or a follow up (your choice) it'd be most appreciated, but no worries if not; this is going to be a great enhancement!

Reply:
I can do a follow-up PR :)

I've created #108 to start a discussion on how we can implement that.